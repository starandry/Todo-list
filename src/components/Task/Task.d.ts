import React from 'react'
interface TaskProps {
    id: string
    title: string
    description: string
    completed: boolean
    date: string
    daysSpent: number
    isFrozen: boolean
    onToggle: (id: string, completed: boolean, date: string, isFrozen: boolean) => void
    onDelete: (id: string) => void
}
declare const Task: React.FC<TaskProps>
export default Task
