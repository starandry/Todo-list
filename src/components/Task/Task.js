import { jsx as _jsx, jsxs as _jsxs } from 'react/jsx-runtime'
import styles from './Task.module.scss'
const Task = ({ id, title, description, completed, date, daysSpent, isFrozen, onToggle, onDelete }) => {
    return _jsxs('div', {
        className: `${styles.task} ${completed ? styles.completed : ''}`,
        children: [
            _jsx('h3', { children: title }),
            _jsx('p', { children: description }),
            _jsxs('p', { className: styles.date, children: ['Created on: ', date] }),
            _jsxs('p', { className: styles.daysSpent, children: ['Days spent on this task: ', daysSpent] }),
            ' ',
            _jsxs('div', {
                className: styles.actions,
                children: [
                    _jsx('button', {
                        onClick: () => onToggle(id, completed, date, isFrozen),
                        children: completed ? 'Undo' : 'Complete',
                    }),
                    _jsx('button', { onClick: () => onDelete(id), children: 'Delete' }),
                ],
            }),
        ],
    })
}
export default Task
