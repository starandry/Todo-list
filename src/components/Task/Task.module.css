.task {
    border-bottom: 1px solid #ddd;
    padding: 10px 0;
}
.task.completed {
    background-color: #e0ffe0;
    color: #555;
}
.task.completed h3 {
    text-decoration: line-through;
}
.task .actions button {
    margin-right: 10px;
    padding: 5px 10px;
    border: none;
    border-radius: 4px;
    cursor: pointer;
}
.task .actions button:first-child {
    background-color: #007bff;
    color: white;
}
.task .actions button:first-child:hover {
    background-color: #0056b3;
}
.task .actions button:last-child {
    background-color: #dc3545;
    color: white;
}
.task .actions button:last-child:hover {
    background-color: #c82333;
}
.task .date {
    font-size: 12px;
    color: #888;
    margin-top: 5px;
}

/*# sourceMappingURL=Task.module.css.map */
