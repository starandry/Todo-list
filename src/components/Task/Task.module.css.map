{"version":3,"sourceRoot":"","sources":["Task.module.scss"],"names":[],"mappings":"AAAA;EACI;EACA;;AAEA;EAII;EACA;;AAJA;EACI;;AAOJ;EACI;EACA;EACA;EACA;EACA;;AAEA;EACI;EACA;;AAEA;EACI;;AAIR;EACI;EACA;;AAEA;EACI;;AAMhB;EACI;EACA;EACA","file":"Task.module.css"}