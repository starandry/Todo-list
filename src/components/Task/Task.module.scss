.task {
  border-bottom: 1px solid #ddd;
  padding: 10px 0;
  transition: background-color 0.3s ease, color 0.3s ease; /* Плавный переход для цвета и фона */

  &.completed {
    h3 {
      text-decoration: line-through;
    }
    background-color: #e0ffe0; /* Задний фон для выполненной задачи */
    color: #555; /* Цвет текста для выполненной задачи */
  }

  .actions {
    button {
      margin-right: 10px;
      padding: 5px 10px;
      border: none;
      border-radius: 4px;
      cursor: pointer;

      &:first-child {
        background-color: #007bff;
        color: white;

        &:hover {
          background-color: #0056b3;
        }
      }

      &:last-child {
        background-color: #dc3545;
        color: white;

        &:hover {
          background-color: #c82333;
        }
      }
    }
  }
}
