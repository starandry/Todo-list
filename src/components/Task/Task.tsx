import React from 'react'
import styles from './Task.module.scss'

interface TaskProps {
    id: string
    title: string
    description: string
    completed: boolean
    date: string
    daysSpent: number
    onToggle: (id: string, completed: boolean) => void
    onDelete: (id: string) => void
}

const Task: React.FC<TaskProps> = ({ id, title, description, completed, date, daysSpent, onToggle, onDelete }) => {
    return (
        <div className={`${styles.task} ${completed ? styles.completed : ''}`}>
            <h3>{title}</h3>
            <p>{description}</p>
            <p className={styles.date}>Created on: {date}</p>
            <p className={styles.daysSpent}>Days spent on this task: {daysSpent}</p> {/* Отображаем количество дней */}
            <div className={styles.actions}>
                <button onClick={() => onToggle(id, completed)}>{completed ? 'Undo' : 'Complete'}</button>
                <button onClick={() => onDelete(id)}>Delete</button>
            </div>
        </div>
    )
}

export default Task
