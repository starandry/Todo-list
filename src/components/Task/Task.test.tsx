import { render } from '@testing-library/react'
import Task from './Task'

describe('Task Component', () => {
    it('renders correctly', () => {
        const mockOnToggle = jest.fn()
        const mockOnDelete = jest.fn()

        const { asFragment } = render(
            <Task
                id="1"
                title="Test Task"
                description="This is a test task"
                completed={false}
                date="2024-10-10"
                daysSpent={5}
                isFrozen={false}
                onToggle={mockOnToggle}
                onDelete={mockOnDelete}
            />
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
