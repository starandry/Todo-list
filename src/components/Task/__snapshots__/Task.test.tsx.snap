// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="task "
  >
    <h3>
      Test Task
    </h3>
    <p>
      This is a test task
    </p>
    <p
      class="date"
    >
      Created on: 2024-10-10
    </p>
    <p
      class="daysSpent"
    >
      Days spent on this task: 5
    </p>
     
    <div
      class="actions"
    >
      <button>
        Complete
      </button>
      <button>
        Delete
      </button>
    </div>
  </div>
</DocumentFragment>
`;
